pyomo/common/tests/test_fileutils.py

```diff
@@ -108,17 +108,16 @@ def test_findfile(self):
         open(os.path.join(subdir,fname),'w').close()
         open(os.path.join(subdir,'aaa'),'w').close()
         # we can find files in the CWD
-        self.assertEqual(
-            os.path.join(self.tmpdir,fname),
-            find_file(fname)
+        self.assertIn(
+            os.path.join(self.tmpdir,fname), find_file(fname)
```

Comment:
In OSX, the path name differs by a one extra word that doesn't happen on Ubuntu/Windows. I.e., instead of the filepath being "/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/", it's "private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/" on MacOSX.

Reply:
Can you make the test:
```
# On OSX, the identified path is prepended with 'private/'
self.assertTrue(find_file(fname).endswith(os.path.join(self.tmpdir,fname)),
                "%s does not end with %s" % (find_file(fname), os.path.join(self.tmpdir,fname)))
```

(and the same for the other test changes?  ... although it is probable sufficient that the comment only appears once.)  Alternatively, can you just add the '`private`' to the `self.tmpdir`?

Reply:
I think your first option might work better. The `private` doesn't appear for other operating systems so changing `self.tmpdir` to add `private` runs the risk of breaking Ubuntu/Windows. I'll test!

Reply:
I changed all of the invocations of "assertIn" to match your suggestion, and they have all passed for both Ubuntu and Mac.

Reply:
Do you have a feel for why `tempfile.mkdtemp()` doesn't include the `private/` leading path, whereas `find_file()` adds it?  That strikes me as slightly concerning.  Does private show up when testing locally, or just in github workflows?  It almost sounds like there is something weird with symbolic links on the github osx build hosts?  (I would have thought that if this was an issue on OSX in general, one of the developers would have come across it and complained.  @michaelbynum, @carldlaird: have either of you seen this before?

Reply:
I can tell you that on my machine (which is a Mac), I have the same issue. For example, if I run `est.pyomo -v --cat="nightly" pyomo/common/tests/test_fileutils.py`:

`======================================================================
FAIL: test_findfile (pyomo.common.tests.test_fileutils.TestFileUtils)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/mmundt/opt/anaconda3/lib/python3.7/site-packages/Pyomo-5.6.7-py3.7.egg/pyomo/common/tests/test_fileutils.py", line 113, in test_findfile
    find_file(fname)
AssertionError: '/var/folders/j7/c891zdjd00d1f3v_k56cbk7r002lg2/T/tmp42kvkn4p/foo.py' != '/private/var/folders/j7/c891zdjd00d1f3v_k56cbk7r002lg2/T/tmp42kvkn4p/foo.py'
- /var/folders/j7/c891zdjd00d1f3v_k56cbk7r002lg2/T/tmp42kvkn4p/foo.py
+ /private/var/folders/j7/c891zdjd00d1f3v_k56cbk7r002lg2/T/tmp42kvkn4p/foo.py
? ++++++++`

Reply:
OK.  Can you look on your machine and see if those two paths actually point to the same file?

Reply:
Looks like they do. It's almost like Macs make two copies.

Reply:
That makes me feel better (it is probably hard link magic).  I think this is a fine solution, although (upon further thought) the following may be a little more clear:
```python
        # On OSX, the identified path is prepended with '/private/'
        # These two paths are in fact equivalent on OSX.
        ref=os.path.join(self.tmpdir,fname)
        found=find_file(fname)
        self.assertTrue(
            found.endswith(ref), "%s does not end with %s" % (found, ref))
```